Fixes on tasks.py, project-updaters and Github actions pipelines

andreriesco opened the pull request:
If the same env var appeared in a task and in a dependsOn task of this task, like the VSCODE_CMD on tcb-platform-push-ostree, with the tcb-unpack as dependsOn, the env var was not getting updated due to it's order on the recursive function. This commit fixes it.

Commit:
project-updater.xsh: Fix _git_status check if is a repo

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
project-updater.ps1: Check if files exist before include them

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
project-updater.xsh: Update also metadata.json on the project

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
project-updater.ps1: Save fixed torizonOSMajor on proj metadata.json

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
github actions: Update actions commands to v4

v4 is the currente version of Github actions commands, and some
commands of the v3 have started to get deprecated, like the artifact
commands:
https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby commented on scripts/project-updater.xsh:
```diff
@@ -151,7 +151,7 @@ if not second_run:
     _git_status = !(git -C @(project_folder) status)
 
-    if "fatal: not a git repository" in _git_status.err:
+    if _git_status.err != None and "fatal: not a git repository" in _git_status.err:
```
in which case the `_git_status.err` is `None`?

andreriesco replied:
When `git -C @(project_folder) status` is a git repository

microhobby commented on scripts/projectUpdater.ps1:
```diff
@@ -1,10 +1,6 @@
 param()
 
-# include
-. "$env:HOME/.apollox/scripts/utils/formatJson.ps1"
```
instead check if the file exists, then include then

andreriesco replied:
Those files were removed from this repo on TOS7, so this files will never exist

microhobby replied:
Can occur if the user change the repo, branch, tag. Is safer check if they exists then include.

andreriesco replied:
In this cases it may happen, I will add the check

andreriesco replied:
Added

microhobby commented on scripts/utils/pip/torizon_templates_utils/tasks.py:
```diff
@@ -927,4 +927,3 @@ def run_task(self, label: str) -> None:
         if _ret.returncode != 0:
             print(f"> TASK [{label}] exited with error code [{_ret.returncode}] <", color=Color.RED)
             raise RuntimeError(f"Error running task: {label}")
-
```
I do not get the meaning of this change

andreriesco replied:
This was a bug present already in [TOS6](https://github.com/torizon/vscode-torizon-templates/pull/294/commits/597c01804ee0ae7795858a249cbe2493401eb1b6), it's the same fix but on the xonsh script. 
But as some other things need to be solved on the pipeline for me to test it on TOS7, I will remove from this MR and add it back on another MR when I have those things fixed

microhobby replied:
ok, this I got it. But I'm meaning of this line in particular 930

microhobby replied:
oh ok, you removed the commit.

Commit:
setup-xonsh.sh: Fix script

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
build-applications.yml: Fix app Github Actions pipeline

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
tasks.py:Use bash to execute the VSCode shell tasks commands and scripts

use bash to execute the VSCode tasks commands and scripts, as they are
written and tested in bash. Valid just for commands of shell type, not
process type ones.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
.gitignore: Fix missing !.github/workflows/build-* exception

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby commented on scripts/bash/setup-xonsh.sh:
```diff
@@ -13,6 +13,7 @@ fi
 set -e
 
 pipx install xonsh
+pipx ensurepath
```
what is this fixing?

andreriesco replied:
This was a suggestion from the `pipx install xonsh`, which suggested adding this to add xonsh to the `PATH` variable:
```
installing xonsh...
⚠️  Note: '/opt/pipx_bin' is not on your PATH environment variable. These apps
    will not be globally accessible until your PATH is updated. Run `pipx
    ensurepath` to automatically add it, or manually modify your PATH in your
    shell's config file (e.g. ~/.bashrc).
 ```

microhobby replied:
this need some kind of password or privilegies?

microhobby replied:
ok, so, I check and this does not need privilegies, so it's ok to have it here.

microhobby commented:
LGTM, there's only these two comments to clarify.
